api/src/main/java/com/sanction/lightning/models/FacebookPost.java

```diff
@@ -1,5 +1,235 @@
+import java.util.List;
+import java.util.Map;
+
 public class FacebookPost {
 
+  @Facebook("id")
+  private String facebookId;
+
+  @Facebook("caption")
+  private String caption;
+
+  @Facebook("created_time")
+  private Date createdTime;
+
+  @Facebook("description")
+  private String description;
+
+  @Facebook("from")
+  private Object from;
+
+  @Facebook("link")
+  private String link;
+
+  @Facebook("message")
+  private String message;
+
+  @Facebook("message_tags")
+  private Map<String, List<Object>> messageTags;
+
+  @Facebook("name")
+  private String name;
+
+  @Facebook("object_id")
+  private String attachedMedia;
+
+  @Facebook("picture")
+  private String picture;
+
+  @Facebook("place")
+  private Object place;
+
+  @Facebook("properties")
+  private List<Object> attachedMediaProperties;
+
+  @Facebook("shares")
+  private Object shares;
+
+  @Facebook("source")
+  private String mediaSource;
+
+  @Facebook("status_type")
+  private String statusType;
+
+  @Facebook("story")
+  private String story;
+
+  @Facebook("to")
+  private List<Object> to;
+
+  @Facebook("updated_time")
+  private Date updatedTime;
+
+  public FacebookPost() {
+
+  }
+
+  @JsonProperty
+  public String getId() {
+    return facebookId;
+  }
+
+  @JsonProperty
+  public String getCaption() {
+    return caption;
+  }
+
+  @JsonProperty
+  public Date getCreatedTime() {
+    return createdTime;
+  }
+
+  @JsonProperty
+  public String getDescription() {
+    return description;
+  }
+
+  @JsonProperty
+  public Object getfrom() {
+    return description;
+  }
+
+  @JsonProperty
+  public String getLink() {
+    return link;
+  }
+
+  @JsonProperty
+  public String getMessage() {
+    return message;
+  }
+
+  @JsonProperty
+  public Map<String, List<Object>> getMesageTags() {
+    return messageTags;
+  }
+
+  @JsonProperty
+  public String getName() {
+    return name;
+  }
+
+  @JsonProperty
+  public String getAttachedMedia() {
+    return attachedMedia;
+  }
+
+  @JsonProperty
+  public String getPicture() {
+    return picture;
+  }
+
+  @JsonProperty
+  public Object getPlace() {
+    return place;
+  }
+
+  @JsonProperty
+  public List<Object> getAttachedMediaProperties() {
+    return attachedMediaProperties;
+  }
+
+  @JsonProperty
+  public Object getShares() {
+    return shares;
+  }
+
+  @JsonProperty
+  public String getMediaSource() {
+    return mediaSource;
+  }
+
+  @JsonProperty
+  public String getStatusType() {
+    return statusType;
+  }
+
+  @JsonProperty
+  public String getStory() {
+    return story;
+  }
+
+  @JsonProperty
+  public List<Object> getTo() {
+    return to;
+  }
+
+  @JsonProperty
+  public Date getUpdateTime() {
+    return updatedTime;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+
+    if (!(obj instanceof FacebookPost)) {
+      return false;
+    }
+
+    FacebookPost other = (FacebookPost) obj;
+    return Objects.equal(this.name, other.name)
```

Comment:
Do we need to use all of them in the `equals` and `hashCode`? Maybe we can determine equality with fewer of the properties. But maybe not, let me know what you think.